custom_navbutton_completion.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Created by ishani on 01/03/2017.
+ */
+M.custom_navbutton_completion = {};
+
+M.custom_navbutton_completion.init = function(Y) {
```

Comment:
Please can you namespace this within M.block_navbuttons, as requested, rather than directly into M?
There aren't currently any other bits of javascript being attached to M.block_navbuttons, but it is much neater to prepare for that now.
As a reminder, I suggested:
M.block_navbuttons = M.block_navbuttons || {};
M.block_navbuttons.completion = {
  init: function(Y) {
  ...
  }
}

Or, if you want to keep it more like the code you have:
M.block_navbuttons = M.block_navbuttons || {};
M.block_navbuttons.completion = {}
M.block_navbuttons.completion.init = function(Y) {
...
}